pinot-common/src/main/java/org/apache/pinot/common/function/FunctionUtils.java

```diff
@@ -126,7 +126,6 @@ private FunctionUtils() {
     put(float[].class, ColumnDataType.FLOAT_ARRAY);
     put(double[].class, ColumnDataType.DOUBLE_ARRAY);
     put(String[].class, ColumnDataType.STRING_ARRAY);
-    put(Object.class, ColumnDataType.OBJECT);
```

Comment:
Removing this leads it to return `null`; hence, we set the `_resultMetadata` in `ScalarTransformationFunction` or in `PostAggregationFunction` as String. Eventually, it fails while evaluating the valueblock.

@Jackie-Jiang / @walterddr do you guys think determining the `_resultMetadata` based on the transformation function  result makes sense like we are doing in [CaseTransformFunction](https://github.com/apache/pinot/blob/master/pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java#L105)